### PR TITLE
feat(dockerstats): add online cpu and memory failcnt metrics

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
**Description:** two new metrics have been added, they were already being scraped but not emitted

**Link to tracking Issue:** #31366

**Testing:** added new metrics to `TestScrapeV2` 

**Documentation:** generated by mdatagen

### CODE REVIEW

File: receiver/dockerstatsreceiver/metadata.yaml
Comment (jamesmoessis) [inline receiver/dockerstatsreceiver/metadata.yaml:153]: What's the use case for this metric that can't be covered by counting MTS of `container.cpu.usage.percpu`?

File: receiver/dockerstatsreceiver/metadata.yaml
Comment (jamesmoessis) [inline receiver/dockerstatsreceiver/metadata.yaml:473]: For new metrics I prefer them not to be enabled by default since this will break downstreams. We already have other memory metrics enabled by default that can indicate memory limits being hit, so I'd probably prefer this to be off-by-default at least initially.

File: receiver/dockerstatsreceiver/metadata.yaml
Comment (jamesmoessis) [inline receiver/dockerstatsreceiver/metadata.yaml:156]: this should be `"{cpus}"` aligning with other metrics here

**Review by jamesmoessis — CHANGES_REQUESTED** [review]
The metric names look good to me. 

You will need to write those tests you mentioned in the description. 

Just a couple of comments but no major issues. Thankyou for the contribution

File: receiver/dockerstatsreceiver/metadata.yaml
Comment (aboguszewski-sumo) [reply receiver/dockerstatsreceiver/metadata.yaml:473]: This is actually an overlook, I totally agree.

**Review by jamesmoessis — APPROVED** [review]
LGTM 👍 

Just needs an approver/maintainer to look at it now
cc @dmitryax as the assignee

**Comment by aboguszewski-sumo** [comment]
Updated to main to fix the failing CI. 
Waiting for a maintainer to approve and merge 🙏

File: receiver/dockerstatsreceiver/documentation.md
Comment (dmitryax) [inline receiver/dockerstatsreceiver/documentation.md:311]: Is there any other documentation available from docker? It's hard to understand what this metric is needed for from this line.

Comment (aboguszewski-sumo) [reply receiver/dockerstatsreceiver/documentation.md:311]: I can't find any documentation from Docker, only some references from the internet that says how people interpret this:
"The number of real used CPUs is given by cpu_stats.online_cpus (4 in our case)." - https://github.com/influxdata/telegraf/issues/2964
"The online_cpus value represents the number of CPU core on the host machine." - https://cratedb.com/blog/analyzing-docker-container-performance-native-tools
Even the source code doesn't say much: https://github.com/moby/moby/blob/460b4aebdf81512a3ea37c2a65826b890427537d/api/types/stats.go#L50

Comment (aboguszewski-sumo) [reply receiver/dockerstatsreceiver/documentation.md:311]: Digging deeper into the code, this is the "cpu_num" return value here: https://github.com/moby/moby/blob/460b4aebdf81512a3ea37c2a65826b890427537d/daemon/stats_unix.go#L321
If we go to the `man` page in the comment, we can see that the command lists cpus with a number like `cpu0`, `cpu1` etc. and Docker just sums number of these different values.

Comment (dmitryax) [reply receiver/dockerstatsreceiver/documentation.md:311]: Ok, it seems like just a number of cores available to the container. We have `system.cpu.logical.count` for this purpose on the host, see https://github.com/open-telemetry/semantic-conventions/blob/6c79a1d9107c1833d95177beda84947ce1902792/model/metrics/system-metrics.yaml#L90C18-L90C42. Should we call it `container.cpu.logical.count`? I think _online_ here just adds confusion.  cc @open-telemetry/semconv-container-approvers

Comment (ChrsMark) [reply receiver/dockerstatsreceiver/documentation.md:311]: It would make sense to introduce it as `container.cpu.logical_count` to be aligned with `system.*`, as per our discussions at https://github.com/open-telemetry/semantic-conventions/pull/282#issuecomment-1909845087.

Comment (aboguszewski-sumo) [reply receiver/dockerstatsreceiver/documentation.md:311]: I have changed the name to `container.cpu.logical_count` as the suggestion said.

Comment (aboguszewski-sumo) [reply receiver/dockerstatsreceiver/documentation.md]: Changed, thanks for responding quickly.

Comment (aboguszewski-sumo) [reply receiver/dockerstatsreceiver/documentation.md]: @dmitryax could you rereview? 🙏

**Comment by aboguszewski-sumo** [comment]
@dmitryax ping :)

**Review by ChrsMark — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
